File: 0027-remove-element/0027-remove-element.go
func removeElement(nums []int, val int) int {
    ans := 0
    index := 0

    for index < len(nums) {
        if nums[index] != val {
            nums[ans] = nums[index]
            ans += 1
        }
        index += 1
    }

    return ans
}

File: 0027-remove-element/0027-remove-element.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        ans = 0
        index = 0

        while index < len(nums):
            if nums[index] != val:
                nums[ans] = nums[index]
                ans += 1
            index += 1
        return ans